lib/puppet/provider/katello_ssl_tool.rb

```diff
@@ -18,6 +18,16 @@ def create
       deploy!   if deploy?
     end
 
+    def destroy
+      files_to_deploy.each do |file|
+        File.delete(file) if File.exist?(file)
+      end
+
+      if(system("rpm -q #{rpmfile_base_name} &>/dev/null"))
```

Comment:
Why is a call to `system()` needed? From reading the Puppet code I think this should work
```suggestion
      output = execute([:rpm, '-q', rpmfile_base_name], failonfail: false)
      if output.exitstatus == 0
```

Reply:
Thanks for that, to be honest I had largely copied the previous code from another spot in the file and the code in here is many, many years old.